### PR TITLE
Redesign start page for clarity and completeness.

### LINKED ISSUE #1761: Add more features to the start page
At the moment, the start page seems to be very blank. It's just a white page and some links to the website, forum etc.

I propose that we add a few things to make the start page a bit more livelier.

- (Minor suggestion) Add the nyxt logo on the start page and in the top left icon box on the window
- Add a search bar that links straight to a search engine? Like with Chrome and Firefox. Or even perhaps just a search bar to input URLs
- Again this is kind of like in Firefox and Chrome, list some commonly used links e.g. Facebook, Youtube etc or even just the last 5 URLs that were opened?

I'm happy to hear any thoughts on this.

### PR DESCRIPTION
**Pull request by aartaka**
This is a minor redesign of a start page to make it clearer what to do when first opening Nyxt. Let the screenshots tell for themselves:

Default theme:
start-page-light

Nyxt Academy dark theme:
start-page-dark

# What's New
- Clear and noticeable call to action right at the top -- for those who want to browse the Internet right away.
- "Documentation" reordered to put "Tutorial" and "Manual" first, as the most often needed materials.
- "Configuration" moved to below "Documentation", as something used less often.
- "Useful Links" section with every other link one may need when using Nyxt.

# Things to Discuss
- I'm itching to remove or collapse everything into icons and dropdowns ("Configuration" could just be a small gear icon on the right, "Documentation" -- question mark somewhere around too). It'll look fabulous, but will it be usable? I'm hesitant here.
- So we actually want to make new users jump into action before configuring and reading about Nyxt? Isn't this Search button too provocative?
- What I wanted to do was a search textarea, but doing it uncovered an interesting feature of web forms: they pass query parameters to the requested URL unquoted, thus driving Lisp reader of `nyxt:` URLs mad. 
  - In other worlds, the format of URL query params is ambiguous and we need to do something about it to make Nyxt fully integratable with it. I see two options:
    - Only accepting strings as `nyxt:` URL arguments (#2160).
    - Somehow imply strings by default and quess other types in exceptional cases. 
  - This is definitely not a start page-related discussion, though.

How do you like this design?

Closes #1761.

EDIT: Refer ot #2160.

### CODE REVIEW

**Comment by Ambrevar** [comment]
Nice, I like it!

**Comment by Ambrevar** [comment]
Some buttons like the tutorial and manual ones have a weird leading spacing, is this expected?

**Comment by Ambrevar** [comment]
Discussion:

- Adding icons is fine, but considering this is a page and not part of the UI, screen real-estate is not important.  So what about adding icons before the text?  The benefit of text is that you can search it and that it's unambiguous.

- I'm fine with the search button being at the top.

- OK with strings as default.

**Comment by pdelfino** [comment]
Great job, Artyom. My only suggestion would be adding a button to "[Nyxt Academy](https://www.youtube.com/channel/UC11mZYESUvaKFTaa3zZWHMQ)" in the useful links section. 

I like the search button. Most people will jump to actions anyways. It does not seem to be provocative in a bad way.

On the icons topic, since this is suppose to help new users, I think text is better. The best starting guides I can remember where **really** explicit. Icons are somewhat ambiguous or even obscure when you have narrow (or no) background knowledge.

Maybe Pierre' suggestion is the best option: adding icons along the text.

**Comment by aadcg** [comment]
Regarding dropdown menus, it could be an option but it would defeat the use of hints.

I second Pierre's observation regarding leading space.

Some suggestions:
- Add Nyxt's logo (centered)
- Remove the smile ":-)"
- Rethink the welcoming message ("Welcome to Nyxt") since the logo might be enough
- I dislike the motto "explore the Internet with your power browser" (see #2166)

Cool Artyom!

**Review by pdelfino — APPROVED** [review]

**Comment by aartaka** [comment]
@aadcg, yes, the "power browser" bit might be a bit over the top.

The suggestion on centered logo brings me to a yet another thing -- do we maybe want to have a centered view for the `nyxt:help`, much like e.g., Firefox, Chrom(e|ium), and everyone else does? It looks fabulous, but it not exactly in line with the other interfaces we have, is it? 

I'm leaning towards centering the content, as it looks more familliar to everyone, from newbies to seasoned users (that are likely to come from Firefox). It's also much easier to make a symmetric and aesthetically pleasing interface in :D

**Comment by aartaka** [comment]
> Some buttons like the tutorial and manual ones have a weird leading spacing, is this expected?

This is because they are laid out in a table. It was there since the dawn of times :D

**Comment by aadcg** [comment]
@aartaka, I realized now that the start page is obviously designed with `help-mode`. I'm not sure about centering all content of every help buffer, but perhaps there could be a `centered-mode` for plain html content. This is also related to how we will handle windows (as defined by Emacs). It seems that Nyxt today is to be used as a "single" window, meaning that there's only one buffer per window. Nonetheless, I think Nyxt should not be a WM. Unless Nyxt will be a shell (in the sense of being the outer-most layer between the OS and the user). But I digress. 

Aesthetically, we're on the same page!

**Comment by aartaka** [comment]
And here goes yet another round! I like this one a lot, but I am sure there will be objections :p 

# What I did:
- Rename `help` to `new`, as it's no longer a help page, but rather a default new buffer one.
- Collapse all the explanatory text into tooltips — these are quite a familiar UI pattern. Besides, most buttons are clear enough to understand the meaning without a long and cluttery text.
- Move settings into the former "Documentation" section and rename it to "Understand Nyxt and make it yours", as a reasonable call to action.
- Add Nyxt Academy link.

# Screenshots (before Nyxt Academy addition)
nyxt-start-light
nyxt-start-academy
nyxt-dark-start

**Comment by Ambrevar** [comment]
I like it a lot!

Nits:

- I'd refrain from underlining, instead use bold or a bigger font.
- It's "Nyxt", not "Nyxt Browser" :)  I would put our motto underneath, in smaller italics.

**Comment by aartaka** [comment]
The reasons I'm hesitant about including the icon are:
- It's not clear where to look for it (it's the same as with GTK libs and source, as in #2176).
  - `nyxt:` URLs are local, thus we cannot request anything from the web, which means we _have to_ search for the icons on the machine.
- Icons should be adjusted to one's screen, and this is a non-trivial thing. Text is much easier to resize, especially when measured in `em`s or `vh`s.

**Comment by aartaka** [comment]
Yet another iteration. All centered, two groups of buttons brought on two sides of the screen. It's so hard to position something exactly at the bottom or exactly in the center of the page that I'm ready to sacrifce "Other usefult links" :D
start

**Comment by aartaka** [comment]
The copyright idea I understand, yes. Not sure where to put it and what to include, though...

**Comment by aartaka** [comment]
And here's the one I like most. Add the copyright -- and it will be perfect :D

minimal

**Comment by aartaka** [comment]
Here's the final one for today, with copyright and version included :)

copy-ver-start

**Comment by Ambrevar** [comment]
For the logo, you could always include it as base64 for now.

The motto is "The Internet on your terms." (Don't forget the "the".)

**Comment by aartaka** [comment]
Done in 02f3a8fa643955c6ed7c8c1d4ad36e22951b9dbf. Merging then?

**Comment by aartaka** [comment]
(No logo yet, as I feel the current design is good enough already :p)

**Comment by Ambrevar** [comment]
OK with me, maybe wait for @jmercouris input?

**Comment by aartaka** [comment]
Sure thing, I'm in no hurry :)

**Comment by aadcg** [comment]
Great work Artyom!

**Comment by aartaka** [comment]
@jmercouris, I see you're back. Any opinion on this redesign? (Final look screenshots attached)

final-start-default
final-theme-academy
final-start-laconia

**Comment by aadcg** [comment]
Just a minor observation.  Perhaps the copyright should contain "????-2022" rather than "2022".

**Comment by aartaka** [comment]
Hmmmmm, yes, I suppose. What's the date we count copyright from? Since the creation of the project? Since Atlas Engineer registration? Since the rename to Nyxt?

**Comment by aadcg** [comment]
Artyom Bologov ***@***.***> writes:

> Hmmmmm, yes, I suppose. What's the date we count copyright from? Since
> the creation of the project? Since Atlas Engineer registration? Since
> the rename to Nyxt?

That's a question for our notable Elders :)

**Comment by Ambrevar** [comment]
The copyright mention is not needed, everything is already licensed or
otherwise implicit.

We  can specify 2018 as a starting year.

**Comment by jmercouris** [comment]
I hate to be bikeshedding, but I really do not like the horizontal redesign. It gives me Google PTSD. That said, you have made some improvements overall, and your addition of the other set of links was a good one. I would like to have a go at modifying this, I may make a branch of your branch. So, please do not merge yet!

**Comment by Ambrevar** [comment]
> Copyright removed in [bef3404](https://github.com/atlas-engineer/nyxt/commit/bef34040a457462be389e3f934bac57f968e3c2c) then :)

Actually I just meant the "©" sign, the rest (the company name and the date) is fine, since it's informative.

**Comment by aartaka** [comment]
> I hate to be bikeshedding, but I really do not like the horizontal redesign. It gives me Google PTSD. That said, you have made some improvements overall, and your addition of the other set of links was a good one. I would like to have a go at modifying this, I may make a branch of your branch. So, please do not merge yet!

Sure thing! WIll be glad to see your vision of it :)

I can relate to Google PTSD, but it's still quite a brain-preserving design :)

**Comment by aartaka** [comment]
> > Copyright removed in [bef3404](https://github.com/atlas-engineer/nyxt/commit/bef34040a457462be389e3f934bac57f968e3c2c) then :)
> 
> Actually I just meant the "©" sign, the rest (the company name and the date) is fine, since it's informative.

Ahhhhhhhhhh, I didn't get it the first time （/｡＼) Should be fixed now!

**Comment by aartaka** [comment]
@jmercouris, friendly ping :) Any ideas on how to improve from there?

**Comment by aadcg** [comment]
@aartaka, is there something that we'd lose if the start page wouldn't be in `help` mode?

**Comment by aartaka** [comment]
> @aartaka, is there something that we'd lose if the start page wouldn't be in `help` mode?

Nothing, `help-mod` is there by accident :D

**Comment by aadcg** [comment]
Great! Thanks.

**Comment by jmercouris** [comment]
I pushed to the branch, can always revert my changes if we don't like them!

So here's what I've come up with. Run it on your machine for a super cool magical effect ;-)

Screenshot_20220415_230000

**Comment by jmercouris** [comment]
Minor tweaks:
Screenshot_20220415_230636

**Comment by aartaka** [comment]
I don't like it :p Okay, I have to explain myself:

### Premises/reactions
- I am intimidated by seeing too much actionable elements. Even before I read all of them -- if there are more than 10, I am automatically lost. That's how brain works, not only for me, but for most people -- there shouldn't be more than 5-7 actionable elements that look similar.
- The space is there either to be meaningfully filled or the stay meaningfully empty. The tails that button have:
  - Don't bear any meaning.
  - Reinforce an accociation with switch buttons on IOS or with heavily styled `<input type=number>` HTML elements. It's confusing.
- Animations are cool if you want to sell something and stimulate immediate action. Color accent works too, but it's much less intense -- people are used to seeing accented buttons. My personal take is that animations are irritating to me, unless they are triggered by my action -- button hovering, for example. Thus, the button with a flowing animation look real irritating to me :D
- If there are several themes, the icon should be adjusted for those. Firefox provides a transparent background PNG, thus fitting in both light and dark theme. Our icon background is black, only working fine in light theme. If it's a dark theme, it blends with the background (see the attached screenshot).
- (Feel free to mock me, but) I like Google-like designs. While they are often terribly sterile, non-functional, and are trying to think for you, they are intuitive for most people. Our interfaces could use some touch of intuitiveness and clarity. 
  - Actually, Google's Material Design Guidelines can be a good ground to make Nyxt Interfaces based on. More so: theming library that I made is a consequence of reading [Material Design Theming](https://material.io/design/material-theming/implementing-your-theme.html#color) guides. See their [Accessibility Guidelines](https://material.io/design/usability/accessibility.html#layout-and-typography), for example lots of reasonable stuff there -- and some doubles what I've said above.

start-page-blinking

**Comment by aadcg** [comment]
@jmercouris, it actually comes close to what I had in mind (namely the column-based screen organization). Good job!

I'll go through Artyom's valuable insights and comment of them.

**Comment by jmercouris** [comment]
Very good feedback Artyom. I made some changes, but have not addressed all issues.

**Comment by jmercouris** [comment]
I believe I've addressed all issues, please give it another try :-)

**Comment by aartaka** [comment]
Indeed, looks much clearer now! While I still like the conventional design more, I have no science/design-backed complains against this one, except the fact that it's still pretty unfamiliar :p

The border of the image could rather be a CSS like 
``` lisp
(img
 :border-style "solid"
 :border-width "1px"
 :border-radius "1px" ; not necessary, yet pretty
 :border-color theme:text) ; or theme:primary, maybe?
```
so that we don't have to hard-code it.

**Comment by jmercouris** [comment]
> The border of the image could rather be a CSS like
> 
> ```lisp
> (img
>  :border-style "solid"
>  :border-width "1px"
>  :border-radius "1px" ; not necessary, yet pretty
>  :border-color theme:text) ; or theme:primary, maybe?
> ```

Yes, it could be, I had to alter the image to do things, it had to be a square. Normally the PNG has rounded corners preventing us from putting a border without there being a gap. Anyways, I've edited the image now to make the border in CSS only.

File: source/help.lisp [hunk 1]
```diff
@@ -818,8 +818,10 @@ The version number is stored in the clipboard."
     (:style (:raw 
              (theme:themed-css (theme *browser*)
                ("#logo"
-                :width "260px"
-                :height "260px")
+                :width "256px"
+                :height "256px"
+                :border "2px solid white"
```
Comment (aartaka) [inline source/help.lisp:823]: Nooooooo, this is not theme-friendly :D 

That's why I suggested splitting the `:border` properties into three — to allow theme-specific colors highlighting the icon!

**Comment by aadcg** [comment]
I like it!  Not sure what's that thing on the lower left :)

Minor things:
- What about the motto?
- The lower limit of the 6 buttons on the left should match the height of the logo.
- Shouldn't it be moved from `help.lisp`?

File: source/help.lisp (hunk 1 above)
Comment (jmercouris) [reply source/help.lisp]: I've pushed a change to fix this. My original reasoning is that it *must* be white to contrast with the black for a nice border, but I guess it doesn't /have/ to be!

**Comment by jmercouris** [comment]
> I like it! Not sure what's that thing on the lower left :)

probably some caching, on some versions it had this issue, but I fixed it by adding a "href" to the `start-searching` anchor.

> Minor things:
> 
>     * What about the motto?

It's not necessary. Plus, I think Tiesto has it covered. https://www.youtube.com/watch?v=1_4ELAxKrDc

>     * The lower limit of the 6 buttons on the left should match the height of the logo.

Not easily possible. Not with an li exactly, also I don't necessarily agree. What if we add/remove elements from this list?

>     * Shouldn't it be moved from `help.lisp`?

Probably :-)

**Comment by Ambrevar** [comment]
I'm in favour of Artyom's version.

- Centered looks nicer in my opinion.
- It's less clobbered, which in my opinion Nyxt really needs now that we are trying to make it more _accessible_ and _familiar_.

**Comment by jmercouris** [comment]
I believe we'll have to agree to disagree. And since my design fits the functional requirements too, I vote for it :-D

**Comment by aadcg** [comment]
Ok, perhaps this wouldn't make everyone happy anyway but here's a suggestion.

Center the logo (as @aartaka did), while bringing the 2-column button layout (as @jmercouris did) below it.

We should leave some buttons out.

**Comment by aartaka** [comment]
> > I like it! Not sure what's that thing on the lower left :)
> 
> probably some caching, on some versions it had this issue, but I fixed it by adding a "href" to the `start-searching` anchor.

That's a bug with WebKit data encoding format :D Strange, it should already be fixed...
 
> It's not necessary. Plus, I think Tiesto has it covered. https://www.youtube.com/watch?v=1_4ELAxKrDc

Tiesto and Ava Max! I see you're a man of a culture :D

> > ```
> > * The lower limit of the 6 buttons on the left should match the height of the logo.
> > ```
> 
> Not easily possible. Not with an li exactly, also I don't necessarily agree. What if we add/remove elements from this list?

Why not? `margin: auto` should cut it.

**Comment by Ambrevar** [comment]
After some discussion with John and Artyom, here is my take away:

- Nyxt is configurable enough that we can afford to offer multiple start pages, leaving it up to the user to choose which ever they prefer.
- We can conduct a poll to choose which one to use by default.
  (My opinion is that we should go for the one that's the simplest and looks the most familiar.)
- John prefers to use the logo, so Artyom's proposal could be modified to replace the "Nyxt" title with a logo.
  - The logo in John's version takes too much screen real estate: it won't be convenient on small screen / in small windows, as the buttons won't show.  We could reduce it's size.

**Comment by aartaka** [comment]
[The poll is up!](https://discourse.atlas.engineer/t/poll-help-us-choose-the-best-new-buffer-page-for-nyxt/480)

**Comment by aadcg** [comment]
With all due respect to everyone's effort, aren't we getting too carried away by minor aspects? Multiple start-up pages... so what's next? Multiple kitchen sinks? :laughing:

**Comment by Ambrevar** [comment]
Good!  I suggest we move forward and merge this pull request at last :)

Minor last minute nit: when on a narrow screen the buttons vertical space is too small, they overlap some bit.  See this screen capture:
nyxt_centered_startup

**Comment by aartaka** [comment]
Rebased over master, waiting for CI to pass, will merge then.

**Comment by jmercouris** [comment]
The polls were rigged! :-P

As I said to Pierre today, I think the fact that results were so close means that we have not yet fully mastered this design!

That said, both alternatives were a vast improvement over the original design, so that is a success!

**Pull request merged**